### PR TITLE
docs: Update api starting guide

### PR DESCRIPTION
**Pull request by stevenGravy**
- Update to allow using variable
- Change instruction to run create user on auth service for cloud usage
- Update go mod instructions to use version variable
- Use less privileged role for api example

### CODE REVIEW

File: docs/pages/api/getting-started.mdx [hunk 1]
```diff
@@ -66,8 +68,8 @@ To find the pseudoversion appropriate for a go.mod file for a specific git tag,
 run the following command from the `teleport` repository:
 
 ```code
-$ go list -f '{{.Version}}' -m "github.com/gravitational/teleport/api@$(git rev-parse v12.1.0)"
-v0.0.0-20230307032901-49a6de744a3a
+$ go list -f '{{.Version}}' -m "github.com/gravitational/teleport/api@$(git rev-parse v(=teleport.version=))"
```
Comment (zmb3) [inline docs/pages/api/getting-started.mdx:71]: I would keep this as is. The pseudoversion below matches v12.1.0 exactly.

File: docs/pages/api/getting-started.mdx [hunk 2]
```diff
@@ -77,48 +79,56 @@ Create a file called `main.go`, modifying the `Addrs` strings as needed:
 package main
 
 import (
-	"context"
-	"log"
+  "context"
```
Comment (zmb3) [inline docs/pages/api/getting-started.mdx:82]: This code is no longer properly formatted. Please revert.

**Comment by ptgott** [comment]
@stevenGravy Checking in on this one

**Comment by ptgott** [comment]
@stevenGravy Circling back on whether we are still going ahead with this one

**Comment by zmb3** [comment]
Closed due to inactivity.

**Pull request closed without merging**